### PR TITLE
#786: Additional API input validation.

### LINKED ISSUE #786: Enforce value for strings extracted from URls
Just to be safe, we should do some input restriction on fields coming in from the HTTP APIs.

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #786.

### CODE REVIEW

**Comment by itamarst** [comment]
https://gitlab.dotmesh.com/dotmesh/dotmesh/pipelines/26818

File: pkg/validator/validator.go [hunk 1]
```diff
@@ -115,7 +122,32 @@ func IsValidSubdotName(str string) error {
...
+
 // ReplaceUUID replace UUID in string
 func ReplaceUUID(str, replace string) string {
 	return rxUUIDPattern.ReplaceAllString(str, replace)
 }
+
+type ValidatorFunc func(string) error
+
+// Return false (and respond with HTTP error) if the value doesn't validate.
+func EnsureValid(value string, validator ValidatorFunc, resp http.ResponseWriter) bool {
```
Comment (binocarlos) [inline pkg/validator/validator.go]: Small suggestion - could this function be called something that indicates it handles the HTTP response also?

`EnsureValid` feels like it is just checking the validity of the value - naming things is hard though and the best I can come up with is `EnsureValidWithResponse` - not a biggie :-)

Comment (itamarst) [reply pkg/validator/validator.go]: Good point, done.

**Comment by itamarst** [comment]
https://gitlab.dotmesh.com/dotmesh/dotmesh/pipelines/26892 is next test run.

**Pull request merged**